subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -1370,10 +1370,37 @@ void ll_rx_mem_release(void **node_rx)
 		{
 			struct node_rx_sync *se =
 				(void *)((struct node_rx_pdu *)rx_free)->pdu;
+			uint8_t status = se->status;
+
+			/* Below status codes use node_rx_sync_estab, hence
+			 * release the node_rx memory.
+			 */
+			if ((status == BT_HCI_ERR_SUCCESS) ||
```

Comment:
I suggest to add 
```
else {
        LL_ASSERT(1);
}
```
at the end of the `if` statement. 
If there is no assert, not allowed error codes will pass unnoticed.

Reply:
Done.

Reply:
To me it doesn't look resolved: @ppryga do you agree?

Reply:
It looks ok, I've forgotten that the `status` may have `BT_HCI_ERR_OP_CANCELLED_BY_HOST` value.